### PR TITLE
implements creating a project entity in registry

### PR DESCRIPTION
**Pull request by Abhishekvrshny**
Also introduces interfaces and base structs for models, repos with
some refactoring

### CODE REVIEW

File: internal/common/repo.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+	Registry registry.Registry
+}
+
+// Create puts the model in the registry
+func (r Repo) Create(m Model) error {
+	b, err := json.Marshal(m)
+	if err != nil {
+		return err
+	}
+	return r.Registry.Put(basePrefix+m.Prefix()+m.Key(), b)
```
Comment (sunnyagg) [inline internal/common/repo.go:24]: should this be exposed by a method in model ?
getKey(){
return basePrefix+m.Prefix()+m.Key()
}

**Review by sunnyagg — COMMENTED** [review]
add tests :)

File: internal/common/repo.go (hunk 1 above)
Comment (Abhishekvrshny) [reply internal/common/repo.go:24]: I think prefix and key can come from Key() method, but base prefix can be in base class itself?

Comment (Abhishekvrshny) [reply internal/common/repo.go:24]: on the other hand, for list functionality, I just need basePrefix+m.Prefix(), so the current scheme works there. Thoughts?

**Review by sunnyagg — APPROVED** [review]

**Pull request merged**